internal/gatewayapi/contexts.go

```diff
@@ -14,14 +14,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
 	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
 )
 
 // GatewayContext wraps a Gateway and provides helper methods for
 // setting conditions, accessing Listeners, etc.
 type GatewayContext struct {
 	*gwapiv1.Gateway
 
-	listeners []*ListenerContext
+	listeners  []*ListenerContext
+	envoyProxy *egv1a1.EnvoyProxy
```

Comment:
I dont see this being set anywhere

Reply:
It's set in `func (t *Translator) Translate(..)` > `func (t *Translator) attachEnvoyProxy(..)`

https://github.com/envoyproxy/gateway/pull/3532/files#diff-be48a4416dad5f211bed7f15e1cc13e8cdfaa1378a7d5be349d8bb06e4b2054eR169-R171

https://github.com/envoyproxy/gateway/pull/3532/files#diff-be48a4416dad5f211bed7f15e1cc13e8cdfaa1378a7d5be349d8bb06e4b2054eR285